scripts: west_commands: add `pyproject.toml` file

thorsten-klein opened the pull request:
### Why?
It is common that tools like `pytest` or `mypy` can be invoked directly.
Especially `pytest` is helpful for being able to debug tests in some IDE.
This should work without any additional effort.

### Actual
Running pytest python module works:
```
/usr/bin/python3 -m pytest
```
Whereby invokation of pytest executable fails:
```
$ pytest
ImportError while loading conftest '/home/user/GIT/zephyr/scripts/west_commands/tests/conftest.py'.
tests/conftest.py:9: in <module>
    from runners.core import RunnerConfig, FileType
E   ModuleNotFoundError: No module named 'runners
```

To fix this, `PYTHONPATH` must be manually set to contain correct path so that `import runner` works:
```
cd scripts/west_commands
PYTHONPATH=. pytest
pytest -o pythonpath=.
```
or from git root folder:
```
pytest scripts/west_commands/ -o pythonpath=scripts/west_commands
```

### Expected
`pytest` can be invoked directly, e.g.:
```
cd scripts/west_commands
pytest
```

Therefore a config file is necessary for pytest.
In order to avoid too many config files for different tools, a `pyproject.toml` config file can be introduced, which allows configuration of both tools `pytest` and `mypy`.

thorsten-klein committed:
scripts: west_commands: add pyproject.toml file

added pyproject.toml config file which allows to set paths for tools
like mypy or pytest. With this config file it becomes possible to
directly run those tools in a local zephyr tree.

Signed-off-by: Thorsten Klein <thorsten.klein@bshg.com>

carlescufi reviewed (commented):
Any chance we can use a format that also runs `mypy` like `run_tests.py` does?

See https://github.com/zephyrproject-rtos/zephyr/blob/b4556ea92304fa492b58e4f4a1e1fe821a2266a5/scripts/west_commands/run_tests.py#L33

thorsten-klein commented:
> Any chance we can use a format that also runs `mypy` like `run_tests.py` does?
> 
> See
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/b4556ea92304fa492b58e4f4a1e1fe821a2266a5/scripts/west_commands/run_tests.py#L33

It seems that `mypi.ini` is already existing:
https://github.com/zephyrproject-rtos/zephyr/blob/b4556ea92304fa492b58e4f4a1e1fe821a2266a5/scripts/west_commands/mypy.ini

I can try to combine those config files and introduce a `pyproject.toml` file which is supported by both `mypy` and `pytest`:
```
[tool.mypy]
ignore_missing_imports=True
mypy_path = .

[tool.pytest]
pythonpath = .
```

thorsten-klein committed:
scripts: west_commands: rework run_tests.py

reworked run_tests.py script:
- removed obsolete arguments from commands called by this script
- avoid code duplication by using a for-loop

Signed-off-by: Thorsten Klein <thorsten.klein@bshg.com>

pdgendt reviewed (approved):
LGTM, as a side-note, there was another PR https://github.com/zephyrproject-rtos/zephyr/pull/96328 that also introduces a `pyproject.toml` file. It does not appear to get much attention though.

I don't know if we should put the file higher in the hierarchy, but we can start with a smaller scope.

carlescufi reviewed (approved):
Thank you!

marc-hb commented on scripts/west_commands/pyproject.toml:
```diff
@@ -0,0 +1,6 @@
+mypy_path = "."
+
+[tool.pytest.ini_options]
+pythonpath = ["."]
```
Please add a link to the relevant documentation because this is surprisingly complex and cryptic:
https://docs.pytest.org/en/7.1.x/explanation/pythonpath.html
